app/src/main/java/com/pollyannawu/gogolook/data/dataclass/ImageLayoutType.kt

```diff
@@ -0,0 +1,7 @@
+package com.pollyannawu.gogolook.data.dataclass
+
+sealed class ImageLayoutType{
+    data class LinearImage(val data: Hit, val isLinear: Boolean): ImageLayoutType()
+    data class GridImage(val data: Hit, val isLinear: Boolean): ImageLayoutType()
+
+}
```

Comment:
如果用了 sealed class，那 isLinear 好像就不一定需要了
因為你就可以透過 type == LinearImage 來表示 isLinear
如果想要 isLinear 來讓 code 簡潔一點，你可以加在 ImageLayoutType 裡面
```
val isLinear: Boolean
  get() = this is LinearImage
```

Reply:
**Resolve this issue.**
- remove isLinear as constructor in ImageLayout
- isLinear value in MainActivity is only used to classify image's layout type, it would not be pass to ImageAdapter.